enterprise/backend/src/metabase_enterprise/serialization/v2/utils/yaml.clj

```diff
@@ -0,0 +1,46 @@
+(ns metabase-enterprise.serialization.v2.utils.yaml
+  (:require [clojure.java.io :as io])
+  (:import java.io.File
+           java.nio.file.Path))
+
+(def ^:private max-label-length 100)
+
+(defn- truncate-label [s]
+  (if (> (count s) max-label-length)
+      (subs s 0 max-label-length)
+      s))
```

Comment:
1. What happens if we have two labels that have the same first 100 characters? Just wanted to make sure identical labels are fine and won't break anything.
2. With Unicode characters e.g. CJK or emoji 100 characters can end up being 400 (?) bytes, hopefully that's ok

If that is going to cause problems 

https://github.com/metabase/metabase/blob/390d10e595ce9d616989fa9c296af6a80235f1c8/shared/src/metabase/mbql/util.cljc#L438-L532

or 

https://github.com/metabase/metabase/blob/f94d5149e3c84ecf1e10d456abc6ff4017d73250/src/metabase/driver/impl.clj#L213-L266

are some ways I've dealt with similar problems in the past (truncate and then append a unique suffix or hash of the original)

Reply:
It seems like since we're using this for filenames preventing duplicates would be important. We probably want to make sure this won't break on case-insensitive file systems either (not sure if that's something we have to worry about or not -- I'm not sure what the resulting label is supposed to look like)